designs/2019-enable-es2017-by-default/README.md

```diff
@@ -0,0 +1,138 @@
+- Start Date: 2019-02-28
+- RFC PR: https://github.com/eslint/rfcs/pull/16
+- Authors: Teddy Katz
+
+# Enable ES2017 parsing and globals by default
+
+## Summary
+
+This RFC proposes to:
+
+* Add an `es2017` environment which enables all ES2015 and ES2017 globals.
+* Enable the `es2017` environment by default.
+* Change the `es6` environment to no longer enable `parserOptions: { ecmaVersion: 6 }`.
+* Change the default value of `parserOptions.ecmaVersion` in `espree` from `5` to `2017`.
```

Comment:
I'm okay with this because it doesn't affect end-users, though I'm not sure what practical value it has.

Reply:
might as well set it to 2019 if it’s being changed :-)

Reply:
The practical effect would be that if a user's config doesn't specify `ecmaVersion`, it gets parsed as ES2017 by default rather than ES5. (ESLint doesn't actually pass a default `ecmaVersion` to parsers -- the default of `ecmaVersion: 5` in ESLint comes from espree.)

Reply:
eslint could start doing so, however, and probably should anyways to insulate users from espree's choices.

Reply:
The fact that we currently don't is an intentional choice -- see [here](https://github.com/eslint/rfcs/blob/f68162f026875971b7d63cc6b3cd5e0dc98f6d59/designs/2019-enable-es2017-by-default/README.md#set-the-default-ecmaversion-in-linter-rather-than-changing-the-default-in-espree).